app/models.py

```diff
@@ -1411,6 +1411,39 @@ class NotificationStatusTypes(db.Model):
     name = db.Column(db.String(), primary_key=True)
 
 
+class NotificationAllTimeView(db.Model):
+    """
+    WARNING: this view is a union of rows in "notifications" and
+    "notification_history". Any query on this view will query both
+    tables and therefore rely on *both* sets of indices.
+    """
+    __tablename__ = 'notifications_all_time_view'
+
+    id = db.Column(UUID(as_uuid=True), primary_key=True)
```

Comment:
rather than redefining everything, can we just make the class definition extend notification history?

`class NotificationAllTimeView(NotificationHistory)`

keeps it more resilient and reduces duplication. we do similar already with TemplateBase vs Template/TemplateHistory

Reply:
I tried this quickly but it's not straightforward:

```
# inheriting from NotificationHistory
NotificationAllTimeView.__tablename__
'notification_history'
```

I expect there's a way around this but I prefer the non-DRY approach:

1. It's easier to see exactly which columns are in the table.
2. It's conceptually clearer: a NotificationAllTimeView isn't a NotificationHistory; it's its own (combo) thing.
3. The other methods and `__table_args__` don't make sense to inherit.

We could potentially have a `NotificationBase` to parallel `TemplateBase`, although it might be a bit weird for the base class of a notification not to have any kind of `to` field.

(1) has been helpful to me on several occasions so a base class path isn't a clear win. `TemplateBase` does set a precedent for it though. Do you think it's worth moving to that pattern for Notifications?